lib/improver/tests/spotdata/spotdata/test_read_input.py

```diff
@@ -0,0 +1,360 @@
+                    pressure_on_height_levels]),
+                'surface_pressure': CubeList([surface_pressure])
+                }
+
+        self.data_directory = mkdtemp()
+
+        self.cube_path = (self.data_directory +
+                          '/01-temperature_at_screen_level.nc')
+        self.cube_path2 = (self.data_directory +
+                           '/02-temperature_at_screen_level.nc')
+        orography_path = self.data_directory + '/orography.nc'
+        land_path = self.data_directory + '/land_mask.nc'
+        ad_path_temperature = (self.data_directory +
+                               '/temperature_on_height_levels.nc')
+        ad_path_pressure = (self.data_directory +
+                            '/pressure_on_height_levels.nc')
+        ad_path_s_pressure = self.data_directory + '/surface_pressure.nc'
+
+        iris.save(cube, self.cube_path)
+        iris.save(cube2, self.cube_path2)
+        iris.save(orography, orography_path)
+        iris.save(land, land_path)
+        iris.save(temperature_on_height_levels, ad_path_temperature)
+        iris.save(pressure_on_height_levels, ad_path_pressure)
+        iris.save(surface_pressure, ad_path_s_pressure)
+
+        diagnostic_recipe = {
+            "temperature": {
+                "diagnostic_name": "air_temperature",
+                "extrema": True,
+                "filepath": "temperature_at_screen_level",
+                "neighbour_finding": {
+                    "land_constraint": False,
+                    "method": "fast_nearest_neighbour",
+                    "vertical_bias": None
+                    }
+                }
+            }
+
+        self.config_path = self.data_directory + '/spotdata_diagnostics.json'
+        ff = open(self.config_path, 'w')
+        json.dump(diagnostic_recipe, ff, sort_keys=True, indent=4,
+                  separators=(',', ': ',))
+        ff.close()
+
+        self.made_files = [self.cube_path, self.cube_path2, orography_path,
+                           land_path, ad_path_temperature, ad_path_pressure,
+                           ad_path_s_pressure, self.config_path]
+
+        self.cube = cube
+        self.cube2 = cube2
+        self.temperature_on_height_levels = temperature_on_height_levels
+        self.ancillary_data = ancillary_data
+        self.additional_data = additional_data
+        self.time_extract = time_extract
+
+    def tearDown(self):
+        """Remove temporary directories created for testing."""
+        for a_file in self.made_files:
+            Call(['rm', a_file])
+        Call(['rmdir', self.data_directory])
+
+
+class Test_Load(Test_read_input):
+    """Test function used for loading data cubes."""
+
+    def test_single_file(self):
+        """Test loading of a single file as an iris.cube.Cube."""
+
+        expected = self.cube
+        result = Plugin('single_file').process(self.cube_path,
+                                               'air_temperature')
```

Comment:
Changed in places where I am using a single file as input. Otherwise I've left the string to indicate I want a particular diagnostic from multiple cubes rather than allowing one of the cubes to define what I'm looking for.